ravenframework/Optimizers/GeneticAlgorithm.py

```diff
@@ -909,14 +1460,24 @@ def _updateConvergence(self, traj, new, old, acceptable):
       @ Out, converged, bool, True if converged on ANY criteria
     """
     # NOTE we have multiple "if acceptable" trees here, as we need to update soln export regardless
-    if acceptable == 'accepted':
-      self.raiseADebug(f'Convergence Check for Trajectory {traj}:')
-      # check convergence
-      converged, convDict = self.checkConvergence(traj, new, old)
-    else:
-      converged = False
-      convDict = dict((var, False) for var in self._convergenceInfo[traj])
-    self._convergenceInfo[traj].update(convDict)
+    if len(self._objectiveVar) == 1: # This is for a single-objective Optimization case.
+      if acceptable == 'accepted':
+        self.raiseADebug(f'Convergence Check for Trajectory {traj}:')
+        # check convergence
+        converged, convDict = self.checkConvergence(traj, new, old)
+      else:
+        converged = False
+        convDict = dict((var, False) for var in self._convergenceInfo[traj])
+      self._convergenceInfo[traj].update(convDict)
+    else: # This is for a multi-objective Optimization case.
+      if acceptable == 'accepted':
+        self.raiseADebug(f'Convergence Check for Trajectory {traj}:')
+        # check convergence
+        converged, convDict = self.checkConvergence(traj, new, old)
+      else:
+        converged = False
+        convDict = dict((var, False) for var in self._convergenceInfo[traj])
+      self._convergenceInfo[traj].update(convDict)
```

Comment:
I do not see the differences between `if` and `else` statements, I think you do not need to check whether it is single or multi objectives.